internal/handler/favourites.go

```diff
@@ -0,0 +1,296 @@
+	var charts map[uuid.UUID]*model.Chart
+
+	go func() {
+		defer wg.Done()
+		charts = h.chartRepository.GetCharts(chartIDs)
+	}()
+
+	// INSIGHTS
+	var insights map[uuid.UUID]*model.Insight
+
+	go func() {
+		defer wg.Done()
+		insights = h.insightRepository.GetInsights(insightIDs)
+	}()
+
+	wg.Wait()
+
+	// Stitch the responses from the various data sources back together in the
+	// same order as the page of favourite ids was retrieved from the DB
+	result := h.stitchResults(userFavourites, audiences, charts, insights)
+
+	c.IndentedJSON(http.StatusOK, result)
+}
+
+func (*FavouritesHandler) stitchResults(userFavourites favourite.FavouriteEntries,
+	audiences map[uuid.UUID]*model.Audience,
+	charts map[uuid.UUID]*model.Chart,
+	insights map[uuid.UUID]*model.Insight) []model.UserFavourite {
+	result := make([]model.UserFavourite, 0, len(userFavourites))
+	for _, f := range userFavourites {
+		switch f.ResourceType {
+		case "audience":
+			{
+				result = append(result, model.UserFavourite{Description: f.Description, Audience: audiences[f.FavouriteId]})
+			}
+		case "chart":
+			{
+				result = append(result, model.UserFavourite{Description: f.Description, Chart: charts[f.FavouriteId]})
+			}
+		case "insight":
+			{
+				result = append(result, model.UserFavourite{Description: f.Description, Insight: insights[f.FavouriteId]})
+			}
+		}
+	}
+	return result
+}
+
+// Splits the list of user favourite entries by resource type and returns 3 slices of ids
+func (*FavouritesHandler) splitUserFavourites(userFavourites favourite.FavouriteEntries) ([]uuid.UUID, []uuid.UUID, []uuid.UUID) {
+	audienceIDs := make([]uuid.UUID, 0, len(userFavourites))
+	insightIDs := make([]uuid.UUID, 0, len(userFavourites))
+	chartIDs := make([]uuid.UUID, 0, len(userFavourites))
+
+	for _, f := range userFavourites {
+		switch f.ResourceType {
+		case "audience":
+			{
+				audienceIDs = append(audienceIDs, f.FavouriteId)
+			}
+		case "chart":
+			{
+				chartIDs = append(chartIDs, f.FavouriteId)
+			}
+		case "insight":
+			{
+				insightIDs = append(insightIDs, f.FavouriteId)
+			}
+		default:
+			{
+				log.Errorf("unknown userfavourite resource type %v", f.ResourceType)
+			}
+		}
+	}
+	return audienceIDs, insightIDs, chartIDs
+}
+
+// CreateUserFavourite godoc
+//
+// @Summary     Create user favourites
+// @Description creates a favourite for a specified user
+// @Tags        favourites
+// @Param       data body model.UserFavouriteShort true "UserFavouriteShort to create"
+// @Produce     json
+// @Success     201	{object}	model.UserFavourite
+// @Failure     400
+// @Failure     401
+// @Failure     404
+// @Failure     500
+// @Router      /favourites [post]
+func (h *FavouritesHandler) CreateUserFavourite(c *gin.Context) {
+	userId, err := auth.GetUserIDFromToken(c)
+	if err != nil {
+		log.Infof("Error obtaining userid from jwt token: %v\n", err)
+		c.AbortWithStatus(http.StatusUnauthorized)
+		return
+	}
+
+	var newFavourite model.UserFavouriteShort
```

Comment:
Can you think of any problems of using models of repository layer in the API requests and responses ?
e.g `var newFavourite model.UserFavouriteShort`

Reply:
Yes, any changes in the data model would seep into the API, likely as a breaking change.